src/cpp/src/debug_utils.hpp

```diff
@@ -29,3 +33,46 @@ void print_tensor(std::string name, ov::Tensor tensor) {
         print_array(tensor.data<bool>(), tensor.get_size());
     }
 }
+
+std::string join(const std::vector<std::string>& listOfStrings, const std::string delimiter) {
+    std::stringstream ss;
+    auto it = listOfStrings.cbegin();
+    if (it == listOfStrings.end()) {
+        return "";
+    }
+    for (; it != (listOfStrings.end() - 1); ++it) {
+        ss << *it << delimiter;
+    }
+    if (it != listOfStrings.end()) {
+        ss << *it;
+    }
+    return ss.str();
+}
+
+template <typename PropertyExtractor>
+static void read_properties(PropertyExtractor&& property_extractor, std::vector<std::string>& output_configuration_values) {
+    auto key = std::string("SUPPORTED_PROPERTIES");  // ov::supported_properties;
+    std::vector<ov::PropertyName> supported_config_keys;
+    try {
+        ov::Any value = property_extractor(key);
+        supported_config_keys = value.as<std::vector<ov::PropertyName>>();
+    } catch (...) {
+        std::cout << "Exception thrown from OpenVINO when requesting model property: " << key << std::endl;
+        return;
+    }
+
+    for (auto& key : supported_config_keys) {
+        if (key == "SUPPORTED_PROPERTIES")
+            continue;
+        std::string value;
+        try {
+            ov::Any param_value = property_extractor(key);
+            value = param_value.as<std::string>();
+        } catch (...) {
+            std::cout << "WARNING: Exception thrown from OpenVINO when requesting model property: " << key << std::endl;
+            continue;
```

Comment:
Why do you handle exceptions here and above? A thrown exception indicates ov API misuse which should be fixed in the implementation.

To avoid failing when no one asked for `get_model_configuration()`, you can construct the vector every time `get_model_configuration()` is called by getting the model via https://github.com/openvinotoolkit/openvino/blob/b9d98cb32e0c3b1de46eecf2c3221858723a7eb8/src/inference/include/openvino/runtime/infer_request.hpp#L328

Reply:
Removed exception clause.
We do not store the infer_request in the cb_pipe.
I think storing the string vector instead of the infer_request is safer and takes up less memory for every cb_pipe.